### DISCUSSION TITLE
Add VLC player backend

### DISCUSSION THREAD

**Discussion by BlakeB415** [TheWidlarzGroup/react-native-video#4959] [Ideas]
### What problem are you trying to solve?

Some codecs and protocols are not supported on the native players. This includes AV1 and Opus (primarily on iOS).

### Proposed solution

Add VLC/VLCKit as an optional backend for playback.

### Platforms

iOS

**Comment by fredrikburmester** [comment]
I'd also love to see this.

**Comment by Lakston** [comment]
Same, especially since there is no documentation on what codecs are supported, I just tried av1 (webm), av1 (mp4) and h265 (mp4) and none of those worked.

**Comment by fredrikburmester** [comment]
I have a (kinda) working example for iOS in my [repo](https://github.com/fredrikburmester/streamyfin/pull/178) of VLC integration with an expo native module. It's no where near the level of react-native-video but it could act as a starting point..

**Comment by freeboub** [comment]
If you want to develop VLC integration, I can provide you some support.
on android The code base is nearly OK to start integration of a new player.
we have some common feature folder (all the react binding and common stuff) they shall be unchanged and an exoplayer folder which can be removed (or duplicated for vlc).

There are still few code organisation changes to do and in small interface to define, but it is really close to be OK.
iOS is a bit harder.

**Comment by Okelm** [comment]
The dream dev experience API: react-native-video automatically pick the player depending on the codec and there's no difference functionality-wise

**Comment by ejkkan** [comment]
I've just created a new feature request for this as well.
https://github.com/TheWidlarzGroup/react-native-video/issues/4669

It would be amazing to see VLC support.

**Comment by TWG-AI** [comment]
Thanks for the idea — noted for the team to consider.

> I'm an AI assistant, so double-check any specifics I may have mentioned.
